src/components/TreeGrid.ts

```diff
@@ -46,31 +48,30 @@ const wrapContent =
     state: 'closed',
   });
 
-export const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<T, T[]>): Row<T>[] => {
-  const traverseHierarchy = (parent: T, depth: number, previousRows: Row<T>[]): Row<T>[] => {
+export const populateTreeFromRoot = <T extends RowContents>(root: T): Row<T>[] => {
```

Comment:
In general, why is root T and the Rows are of type T, instead of explicitly naming it a SnapshotBuilderConcept which also seems to extends RowContents?

Reply:
This code uses `T` because `TreeGrid` is a general component that doesn't use any dataset builder types. Currently dataset builder is the only user of it but I'm trying to keep it usable by other components too.

By writing `T extends RowContents` that means that a type parameter to `TreeGrid` must conform to `RowContents`. It could do this by extending but typescript uses "duck typing" which means that as long as the type has the required properties, it's conforming to that type, even if it doesn't refer to the other type directly. So `SnapshotBuilderConcept` has the required fields but doesn't actually extend `RowContents`.

Reply:
Thank you!